File: Algorithms/Strings/Separate the Numbers.py
#
# Complete the 'separateNumbers' function below.
#
# The function accepts STRING s as parameter.
#

def separateNumbers(s):
    # Write your code here
    for i in range(1, len(s) // 2 + 1):
        first_num = s[:i]
        current_num = int(first_num)
        remaining = s[i:]
        while remaining:
            next_num = str(current_num + 1)
            if remaining.startswith(next_num):
                current_num = int(next_num)
                remaining = remaining[len(next_num) :]
            else:
                break
        if not remaining:
            print("YES", first_num)
            return
    print("NO")

if __name__ == '__main__':
    q = int(input().strip())

    for q_itr in range(q):
        s = input()

        separateNumbers(s)

File: Algorithms/Strings/Separate the Numbers.php
<?php

/*
 * Complete the 'separateNumbers' function below.
 *
 * The function accepts STRING s as parameter.
 */

function separateNumbers($s) {
    # Write your code here
    $n = strlen($s);
    $isValid = "NO";

    for ($i = 1; $i <= $n / 2; $i++) {
        $firstNum = (int) substr($s, 0, $i);
        $current = $firstNum;
        $num = $firstNum;

        while (strlen($current) < $n) {
            $current .= ++$num;
        }

        if ($current === $s) {
            $isValid = "YES $firstNum";
            break;
        }
    }

    echo $isValid . PHP_EOL;
}

$q = intval(trim(fgets(STDIN)));

for ($q_itr = 0; $q_itr < $q; $q_itr++) {
    $s = rtrim(fgets(STDIN), "\r\n");

    separateNumbers($s);
}
